### PR TITLE
[OpenAPI] Add Documentation for the Java CLI

### PR DESCRIPTION
**Pull request by Johannes-Schneider**
## What Has Changed?

Add usage documentation for the OpenAPI CLI (Java).

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/java/features/rest/generate-rest-client.mdx [hunk 1]
```diff
@@ -175,18 +179,52 @@ As an example refer to the plugin XML below:
 </plugin>
 ```
 
-### Available Parameters
+## Using the OpenAPI Generator CLI
+
+1. Download the latest command-line interface (CLI) of the generator from [Maven Central](https://search.maven.org/artifact/com.sap.cloud.sdk.datamodel/openapi-generator-cli).
```
Comment (Johannes-Schneider) [inline docs/java/features/rest/generate-rest-client.mdx:184]: This artifact has not yet been published to Maven Central. Therefore, don't merge this PR until the artifact can be found.

File: docs/java/features/rest/generate-rest-client.mdx [hunk 2]
```diff
@@ -15,7 +15,11 @@ keywords:
   - generate
 ---
 
-The SAP Cloud SDK offers an OpenAPI client generator as a Maven plugin which can be used to generate a client library for a REST API based on its OpenAPI specification.
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+The SAP Cloud SDK offers an OpenAPI client generator as a Maven plugin and as a command line tool.
```
Comment (artemkovalyov) [inline docs/java/features/rest/generate-rest-client.mdx:21]:
```suggestion
The SAP Cloud SDK offers an OpenAPI client generator as a Maven plugin and as a command-line tool.
```

Comment (artemkovalyov) [reply docs/java/features/rest/generate-rest-client.mdx:21]: For consistency with other occasions of this term.

**Review by artemkovalyov — APPROVED** [review]
Looks good

**Pull request merged**